From 4e06e81db82343b61570c9092e1bc1303b3a31a6 Mon Sep 17 00:00:00 2001
From: vipinc <vipinc@ami.com>
Date: Mon, 11 Sep 2023 12:15:38 +0530
Subject: [PATCH] Add message registry entry for OS Critical Sensor

Add "OSCriticalStop" event message entry in RedFish for
Logging events based on the Event in Operating System.

Tested in Intel Archercity:

1. Verified in RedFish, "OSCriticalStop" events logged properly
GET:
https://<BMC-IP>/redfish/v1/Systems/system/LogServices/EventLog/Entries
{
"@odata.id": "/redfish/v1/Systems/system/LogServices/EventLog/Entries/1694413152",
"@odata.type": "#LogEntry.v1_9_0.LogEntry",
"Created": "2023-09-11T06:19:12+00:00",
"EntryType": "Event",
"Id": "1694413152",
"Message": "Run-time Critical Stop Event is detected in Operating System",
"MessageArgs": [
"Run-time Critical Stop"
],
"MessageId": "OpenBMC.0.1.OSCriticalStop",
"Name": "System Event Log Entry",
"Severity": "OK"
}

Signed-off-by: vipinc <vipinc@ami.com>
---
 .../include/registries/openbmc_message_registry.hpp    | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/redfish-core/include/registries/openbmc_message_registry.hpp b/redfish-core/include/registries/openbmc_message_registry.hpp
index f72fc7e8..19ef08e0 100644
--- a/redfish-core/include/registries/openbmc_message_registry.hpp
+++ b/redfish-core/include/registries/openbmc_message_registry.hpp
@@ -2027,6 +2027,16 @@ constexpr std::array registry = {
             {"string", "string"},
             "None.",
         }},
+    MessageEntry{
+        "OSCriticalStop",
+        {
+            "Indicates that Critical Event in Operating System is Detected",
+            "%1 Event is detected in Operating System",
+            "OK",
+            1,
+            {"string"},
+            "None.",
+        }},
     MessageEntry{"ResetButtonPressed",
                  {
                      "Indicates that the reset button was pressed.",
-- 
2.25.1

